UnityProject/Assets/Scripts/UI/Systems/SplittingMenu.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace UI
+{
+	public class SplittingMenu : MonoBehaviour
+	{
+		[SerializeField]
+		private InputField amountInput;
+
+		public static SplittingMenu Instance;
+		private int AmountToTransfer;
+		private ItemSlot currentSlot;
+		private ItemSlot stackSlot;
+
+		public void Awake()
+		{
+			if (Instance == null)
+			{
+				Instance = this;
+			}
+		}
+
+		public void Enable()
+		{
+			this.SetActive(true);
+		}
+
+		public void Disable()
+		{
+			this.SetActive(false);
+		}
+
+		public void BtnConfirmation()
+		{
+			if (amountInput.text.All(c => (c >= 48 && c <= 57))) //Checking if string is numerical
```

Comment:
you could use try pass 
if (int.TryParse(string, out num))
{
   ...
}

Reply:
Is that faster solution?

Reply:
probably, and it reads better

Reply:
Another option
```suggestion
			if (amountInput.text.All(Char.IsDigit()) //Checking if string is numerical
```
would take care of decimals

Reply:
In this case whitespaces become a problem. Ill try tryparse

Reply:
Fixed in new commit